.github/workflows/diagram-cache.yml

```
@@ -0,0 +1,57 @@
+# Copyright DB Netz AG and contributors
+# SPDX-License-Identifier: Apache-2.0
+
+name: update diagram cache
+
+on: push
+
+env:
+  CAPELLA_DOCKER_IMAGES_REVISION: main
+
+  # Can either be no or selected
+  CAPELLA_DOCKER_IMAGES_DROPINS: selected
+
+  # Semantic Capella version (x.x.x) >= 5.0.0
+  # Defaults to 6.0.0 if not defined
+  CAPELLA_VERSION: 6.0.0
+
+jobs:
+    prepare-image:
+      runs-on: ubuntu-latest
+      steps:
+        - name: Empty step
+          run:
+            echo hallo
+      outputs:
+        image_tag: ${{ env.CAPELLA_VERSION }}-${{ env.CAPELLA_DOCKER_IMAGES_DROPINS }}-dropins-${{ env.CAPELLA_DOCKER_IMAGES_REVISION }}
+
+    update-diagram-cache:
+      runs-on: ubuntu-latest
+      needs: [prepare-image]
+      container:
+        image: ghcr.io/dsd-dbs/capella-dockerimages/capella/base:${{ needs.prepare-image.outputs.image_tag }}
```

```suggestion
        image: ghcr.io/dsd-dbs/capella-dockerimages/capella/base:${{ env.CAPELLA_VERSION }}-${{ env.CAPELLA_DOCKER_IMAGES_DROPINS }}-dropins-${{ env.CAPELLA_DOCKER_IMAGES_REVISION }}
```

Shoudn't this also work? The empty step should be redundant then.

---

I wanted to do it like that but my IDE and also this [action](https://github.com/DSD-DBS/coffee-machine/actions/runs/5412248945) shows you that somehow `env` is not available in that context... according to my IDE only `github`, `inputs`, `vars`, `needs`, `strategy` and `matrix` are available here

---

Indeed, you are right. Do you know if it would work when we define our own action? Then we can also directly go for the action / template :)

---

tbh honest I don't know. Then those parameters would be passed as an input. Looking into [that](https://stackoverflow.com/questions/58476228/how-to-use-a-variable-docker-image-in-github-actions) I does not look possible either but I am not so sure about it.

---

Alternatively, we can use a feature of capellambse: "The CLI of Capella must be specified by the caller. It is possible to work with a local installation or a Docker image of an individual Capella bundle."

We could just take any basic image and then run `docker run` with out image in one of the steps.

---

https://dsd-dbs.github.io/py-capellambse/code/capellambse.model.html#capellambse.model.MelodyModel.update_diagram_cache

---

this [PR](https://github.com/DSD-DBS/py-capellambse/pull/309) should help